nesc/whip6/apps/ICountRead/ICountReadPrv.nc

```diff
@@ -0,0 +1,42 @@
+/*
+ * whip6: Warsaw High-performance IPv6.
+ *
+ * Copyright (c) 2017 University of Warsaw
+ * All rights reserved.
+ *
+ * This file is distributed under the terms in the attached LICENSE
+ * files.
+ */
+
+#define INTERVAL_1S 1024
+
+module ICountReadPrv {
+    uses interface Boot;
+    uses interface Timer<TMilli, uint32_t>;
+    uses interface EventCount as ICount;
+}
+
+implementation {
+    int timestamp;
+
+    event void Boot.booted() {
+        call ICount.start();
+        timestamp = 0;
+        call Timer.startWithTimeoutFromNow(INTERVAL_1S);
+    }
+
+    event void Timer.fired() {
+        unsigned long long ticks = 0;
+        error_t res = SUCCESS;
+
+        timestamp += 1;
+        call Timer.startWithTimeoutFromLastTrigger(INTERVAL_1S);
+
+        res = call ICount.read(&ticks);
+        if (res != SUCCESS)
+            printf("Failed to read ICount\n");
+        else
+            printf("ICount - number of ticks: %u, timestamp: %d\n",
```

Comment:
Currently printf implementation doesn't support these flags (%lu and %llu). See:
nesc/whip6/lib/extern/tfp_printf/cortex-m3/tfp_printf.c